app/models/match_result.rb

```diff
@@ -0,0 +1,41 @@
+# == Schema Information
+#
+# Table name: match_results
+#
+#  id               :integer          not null, primary key
+#  match            :integer          (points to a Match Object)
+#  winner_advances  :boolean
+#
+
+class MatchResult < ApplicationRecord
+  belongs_to :match
+
+  def name
+    if knows_advancing_participant
+      advancing_participant.name
+    else
+      description
+    end
+  end
+
+  def description
+    key = winner_advances ? 'matches.winner_of_match' : 'matches.loser_of_match'
+    I18n.t key, match: match.round
+  end
+
+  def last_match_of(team)
+    match.last_match_of team
+  end
+
+  def knows_advancing_participant
+    advancing_participant.present?
+  end
+
+  def advancing_participant
+    if winner_advances
+      match.winner
```

Comment:
I fear this might result in very costly queries when viewing the page of a single match...

Can we do some caching, if we use the appropriate [active record callbacks](http://guides.rubyonrails.org/active_record_callbacks.html#available-callbacks)?

Reply:
performance gets addressed when needed